### ISSUE TITLE
Improve StartKoinInActivity detection - use PSI instead of string matching

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#54]
## Context
Code review comments from PR #47: https://github.com/androidbroadcast/Koin-Detekt/pull/47

### 1. String matching causes false positives
Current implementation checks if argument text contains "activity" or "fragment" (case-insensitive). This can match unrelated variables:

**False positives:**
- `activityData`
- `fragmentedData`
- `transactionActivity`
- `fragmentManager`

### 2. Supertype detection too broad
Uses `.contains()` for supertype matching:
```kotlin
superTypes.any { it.contains("Application") }
```

This matches unintended types:
- `NetworkApplication`
- `DatabaseApplicationComponent`
- Any class with "Application" in the name

### 3. Misses indirect inheritance
Doesn't detect indirect inheritance through base classes:
```kotlin
class MyActivity : BaseActivity : AppCompatActivity : Activity
```

### For argument detection:
Use stricter pattern matching:
```kotlin
val suspiciousArgument = argument == "activity" ||
        argument == "fragment" ||
        argument.endsWith("Activity") ||
        argument.endsWith("Fragment")
```

Or add import-based detection to verify the parameter type comes from `android.app` or `androidx` packages.

### For supertype detection:
Use exact type name matching:
```kotlin
// Extract simple supertype names (without package or generics)
val superTypes = containingClass.superTypeListEntries
    .mapNotNull { it.typeReference?.text }
    .map { it.substringBefore('<').substringAfterLast('.') }

// Known Android framework entry types
val activityTypes = setOf("Activity", "AppCompatActivity", "FragmentActivity", "ComponentActivity")
val fragmentTypes = setOf("Fragment")

val isActivity = superTypes.any { it in activityTypes }
val isFragment = superTypes.any { it in fragmentTypes }
val isApplication = superTypes.any { it == "Application" }
```

## Priority
IMPORTANT - Causes false positives and false negatives.

## Related
- PR #47: https://github.com/androidbroadcast/Koin-Detekt/pull/47
- Issue #50 tracks general PSI improvements for Group 4 rules
